plugins/inputs/opcua/opcua_client.go

```diff
@@ -210,50 +312,75 @@ func (o *OpcUA) InitNodes() error {
 	return nil
 }
 
+type metricParts struct {
+	metricName string
+	fieldName  string
+	tags       string // sorted by tag name and in format tag1=value1, tag2=value2
+}
+
+func newMP(n *Node) metricParts {
+	var keys []string
+	for key := range n.metricTags {
+		keys = append(keys, key)
+	}
+	sort.Strings(keys)
+	var sb strings.Builder
+	for i, key := range keys {
+		if i != 0 {
+			sb.WriteString(", ")
+		}
+		sb.WriteString(key)
+		sb.WriteString("=")
+		sb.WriteString(n.metricTags[key])
+	}
+	x := metricParts{
+		metricName: n.metricName,
+		fieldName:  n.tag.FieldName,
+		tags:       sb.String(),
+	}
+	return x
+}
```

Comment:
This function might benefit speedwise from the optimization done here: https://github.com/influxdata/telegraf/pull/8391.

Reply:
Since this validation is only needed once during plugin initialization I didn't think optimization was worthwhile